packages/main/src/StepInput.ts

```diff
@@ -367,7 +376,15 @@ class StepInput extends UI5Element implements IFormElement {
 	}
 
 	get _valuePrecisioned() {
-		return this.value.toFixed(this.valuePrecision);
+		if ((this.value === 0 && this._initialRender) || (Number.isInteger(this.value) && !this.valuePrecision)) {
+			return this.value.toFixed(this.valuePrecision);
+		}
+
+		if (this.value === Number(this.input.value)) { // For the cases where the number is fractional and is ending with 0s.
+			return this.input.value;
+		}
+
+		return this.value.toString();
```

Comment:
Is there a case when the value returned by this function isn't precisioned?

Reply:
Most of the times we expect it to not be precisioned. 
The only times we expect it to be precisioned is when working with whole numbers (f.e in cases where `valuePrecision` is not provided, but a `step` property (which is not whole number) is provided, like `step="0.5"`), and after the first render, to illustrate `valuePrecision` property effect. For example to illustrate 0.00 (if we have `valuePrecision="2"`)

Reply:
I see. Before we've always precisioned the value, but now we do in those cases you've described. Not sure if the method name explains it's purpose well. Currently this method will return a string representation of the value, which will be precisioned in one of the cases.  Maybe we could think for a name like "_displayedValue" or "_inputValue".